server/service/src/stocktake/insert.rs

```diff
@@ -121,7 +123,19 @@ fn generate(
         Some(_) => generate_lines_with_stock(connection, store_id, &id)?,
         None => Vec::new(),
     };
-    let lines = [master_list_lines, location_lines, items_have_stock_lines].concat();
+    let expiring_items_lines = match expires_before {
+        Some(_) => {
+            generate_lines_expiring_before(connection, store_id, &id, &expires_before.unwrap())?
```

Comment:
Is this an appropriate use of `unwrap()`? Is there a better way?

Reply:
I guess it can't panic, given you've already matched that `expires_before` is a `Some`, but we'd usually extract the inner value of a Some like this:

```rust
Some(expiry_date) => generate_lines_expiring_before(connection, store_id, &id, &expiry_date)?
```

Reply:
Great, thank you: d6bc95482

🙏